docs/doc_gen_diff.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -eu
+#
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(mbrukman): figure out how to encode this in Makefile syntax for
+# simplicity. In particular, the diff command with subcommand diff does not seem
+# to work and it's unclear what the correct syntax is for this, so keeping this
+# in a shell script for now.
+for md in *.md.in ; do
```

Comment:
Would it make sense to move this into the doc_gen script?  Right now this is only going to work on systems with bash (i.e it's not going to work on Windows).  AFAIK gnumake for Windows doesn't come with bash out of the box, the developer would need to install mingw / cygwin etc.

Reply:
Does Windows Subsystem for Linux provide `bash`? :smile:

I've added a flag `-d` to the `doc_gen.py` script so it can diff the planned output with the contents of the file on dis. I've also added a `-w` flag to overwrite files in-place, so we don't have to redirect stdout (but I kept that option as well). In the future, if/when we have more than one such file, rewriting all `*.md.in` files will be as easy as:

```
./doc_gen.py -w *.md.in
```